### PR TITLE
Add offset to the histogram of the response time

### PR DESCRIPTION
**Pull request by debbyku**
It is to add offsset to the histogram of the response time.

Currently we run the fortio on a service with predefined latency, say, 200ms, 500ms, 1s to simulate production behavior.  Fortio does not give accurate response time as the histogram of the response time is always starting from 0.   

In this PR, a new flag is added.  it is
-o :  offset to histogram of the response time
e.g.  if the latency of the service is 200ms,  i.e. sleep 200ms in the service.  the command for the fortio is : '
  
fortio load -o 0.2s    ......

### CODE REVIEW

File: fortio_main.go [hunk 1]
```diff
@@ -104,6 +104,7 @@ var (
 	durationFlag    = flag.Duration("t", defaults.Duration, "How long to run the test or 0 to run until ^C")
 	percentilesFlag = flag.String("p", "50,75,90,99,99.9", "List of pXX to calculate")
 	resolutionFlag  = flag.Float64("r", defaults.Resolution, "Resolution of the histogram lowest buckets in seconds")
+	offsetFlag      = flag.Float64("o", defaults.Offset, "Offset of the histogram data in seconds")
```
Comment (ldemailly) [inline fortio_main.go]: should probably be a duration (like -t)

Comment (debbyku) [reply fortio_main.go]: Code updated and pushed again

File: periodic/periodic.go [hunk 2]
```diff
@@ -406,7 +408,8 @@ func (r *periodicRunner) Run() RunnerResults {
 	}
 	start := time.Now()
 	// Histogram  and stats for Function duration - millisecond precision
-	functionDuration := stats.NewHistogram(0, r.Resolution)
+	offsetSec := float64(r.Offset) / float64(time.Second)
```
Comment (ldemailly) [inline periodic/periodic.go]: https://pkg.go.dev/time#Duration.Seconds

**Review by ldemailly — CHANGES_REQUESTED** [review]
see above

File: periodic/periodic.go (hunk 2 above)
Comment (debbyku) [reply periodic/periodic.go]: thanks. It is much easier to get the seconds.  Code updated.

**Review by ldemailly — APPROVED** [review]
thanks!

**Comment by ldemailly** [comment]
I should have thought about this sooner, but I am going to change the flag to keep the short `-o` for future

**Comment by ldemailly** [comment]
#496 - so please use 1.18.2 and `-offset` - thanks!

**Comment by debbyku** [comment]
thanks

在 2021年11月24日週三 02:21，Laurent Demailly ***@***.***> 寫道：

**Pull request merged**